workshops/test/test_security.py

```diff
@@ -0,0 +1,254 @@
+        """
+        view_name = 'event_submit'
+        view = get_view_by_name(view_name)
+        assert LoginNotRequiredMixin in view.view_class.__mro__
+        url = reverse(view_name)
+
+        self.assert_accessible(url, user='superuser')
+        self.assert_accessible(url, user='admin')
+        self.assert_accessible(url, user='trainee')
+        self.assert_accessible(url, user=None)
+
+    IGNORED_VIEWS = [
+        'selectable-lookup',
+
+        # auth
+        'login',
+        'logout',
+        'password_reset',
+        'password_reset_done',
+        'password_reset_confirm',
+        'password_reset_complete',
+        'password_change',
+        'password_change_done',
+
+        # python-social-auth
+        'begin',
+        'complete',
+        'disconnect',
+        'disconnect_individual',
+    ]
+
+    def test_all_views_have_explicit_access_control_defined(self):
+        """
+        Test that all views have explicitly defined access control:
+
+        - In the case of function based views, test if the view is decorated
+          with an access control decorator like `login_not_required`.
+
+        - In the case of class based views, test if the view have access control
+          mixin like `LoginNotRequiredMixin`.
+
+        Ignores:
+
+        - views listed on IGNORED_VIEWS,
+
+        - Django-admin views,
+
+        - Rest API views.
+
+        """
+
+        all_views = get_resolved_urls(urls.urlpatterns)
+
+        # ignore views listed on IGNORED_VIEWS list
+        all_views = [v for v in all_views
+                     if not hasattr(v, 'name') or v.name not in self.IGNORED_VIEWS]
+
+        for v in all_views:
+            with self.subTest(view=v):
+                access = getattr(v.callback, 'access', None)
+                model_admin = getattr(v.callback, 'model_admin', None)
+                admin_site = getattr(v.callback, 'admin_site', None)
+
+                # v.callback is always a function, even when the view is
+                # class based. We try to find the class implementing the view.
+                view_class = getattr(v.callback, 'view_class', None)
+                cls = getattr(v.callback, 'cls', None)
+                class_ = view_class or cls or None
+
+                is_function_based_view = (model_admin is None and
+                                          admin_site is None and
+                                          class_ is None)
+
+                if is_function_based_view:
+                    self.assertTrue(access is not None,
+                                    "You have a function based view with"
+                                    "no access control decorator. This view is"
+                                    "probably accessible to every user. If "
+                                    "this is what you want, use "
+                                    "@login_not_required decorator.")
```

Comment:
Additionally I can see inconsistency with using `'` vs `"` for strings.

AMY's style evolved over time and what I now do is:
* `'` for strings everywhere
* except for docstrings which are `"""` in most cases, except if I can manage to fit in one line by dropping `""`, then I'm using `"`

So:
```python
def function():
    """Do this and that."""
    v = 'Test string'
    return v
```